base/bucket.go

```diff
@@ -156,86 +152,36 @@ func (spec BucketSpec) UseClientCert() bool {
 	return true
 }
 
-type GoCBConnStringParams struct {
-	// The KV pool size, KV buffer size and DCP buffer size are passed down to gocbcore.
-	// Defaults to different values based on being in serverless or not.
-	KVPoolSize    int
-	KVBufferSize  int
-	DCPBufferSize int
+// GetGoCBConnString builds a gocb connection string based on BucketSpec.Server.
+func (spec *BucketSpec) GetGoCBConnString() (string, error) {
+	return spec.getGoCBConnString(nil)
 }
 
-// FillDefaults replaces any unset fields in this GoCBConnStringParams with their default values.
-func (p *GoCBConnStringParams) FillDefaults() {
-	if p.KVPoolSize == 0 {
-		p.KVPoolSize = DefaultGocbKvPoolSize
-	}
-	if p.KVBufferSize == 0 {
-		p.KVBufferSize = 0
-	}
-	if p.DCPBufferSize == 0 {
-		p.DCPBufferSize = 0
-	}
+// //GetGoCBConnStringforDCP builds a gocb connection string from BucketSpec.Server for DCP connections.
+func (spec *BucketSpec) GetGoCBConnStringForDCP() (string, error) {
+	return spec.getGoCBConnString(IntPtr(GoCBPoolSizeDCP))
 }
 
-// GetGoCBConnString builds a gocb connection string based on BucketSpec.Server.
-func (spec *BucketSpec) GetGoCBConnString(params *GoCBConnStringParams) (string, error) {
-	if params == nil {
-		params = &GoCBConnStringParams{}
-	}
-	params.FillDefaults()
-	connSpec, err := gocbconnstr.Parse(spec.Server)
+// getGoCBConnString builds a gocb connection string based on BucketSpec.server values. This is used for bucket connections. KvPoolSize can be forced despite the values of the connection values.
+func (spec *BucketSpec) getGoCBConnString(forceKvPoolSize *int) (string, error) {
```

Comment:
Did you consider an approach that would allow us to override other values in future?  (e.g. kv_buffer_size, dcp_buffer_size), or do you think that's unlikely to be necessary?  I was thinking of something like `connStringOverrides map[string]int`, but I can't decide whether that's over-engineering the solution.

Reply:
I think it isn't necessary for now, unless we decide that we need to for dcp_buffer_size in cbgt. I think when we need to expand it, we'd should cross the bridge when we come to it.